dlp/snippets/inspect/inspect_string_custom_hotword.go

```diff
@@ -0,0 +1,124 @@
+// limitations under the License.
+
+package inspect
+
+// [START dlp_inspect_string_custom_hotword]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+	"google.golang.org/api/option"
+)
+
+// inspectStringCustomHotWord inspects a string from sensitive data by using a custom hot word
+func inspectStringCustomHotWord(w io.Writer, projectID, textToInspect, customHotWord, infoTypeName string) error {
+	// projectID := "my-project-id"
+	// textToInspect := "patient name: John Doe"
+	// customHotWord := "patient"
+	// infoTypeName := "PERSON_NAME"
+
+	ctx := context.Background()
+
+	// Initialize a client once and reuse it to send multiple requests. Clients
+	// are safe to use across goroutines. When the client is no longer needed,
+	// call the Close method to cleanup its resources.
+	client, err := dlp.NewRESTClient(ctx, option.WithCredentialsFile("C:/Users/aarsh.dhokai/Desktop/cred.json"))
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the type and content to be inspected.
+	var contentItem = &dlppb.ContentItem{
+		DataItem: &dlppb.ContentItem_ByteItem{
+			ByteItem: &dlppb.ByteContentItem{
+				Type: dlppb.ByteContentItem_TEXT_UTF8,
+				Data: []byte(textToInspect),
+			},
+		},
+	}
+
+	// Increase likelihood of matches that have customHotword nearby
+	var hotwordRule = &dlppb.InspectionRule_HotwordRule{
+
+		HotwordRule: &dlppb.CustomInfoType_DetectionRule_HotwordRule{
+			HotwordRegex: &dlppb.CustomInfoType_Regex{
+				Pattern: customHotWord,
+			},
+			Proximity: &dlppb.CustomInfoType_DetectionRule_Proximity{
+				WindowBefore: 50,
+			},
+			LikelihoodAdjustment: &dlppb.CustomInfoType_DetectionRule_LikelihoodAdjustment{
+				Adjustment: &dlppb.CustomInfoType_DetectionRule_LikelihoodAdjustment_FixedLikelihood{
+					FixedLikelihood: dlppb.Likelihood_VERY_LIKELY,
+				},
+			},
+		},
+	}
+
+	// Construct a ruleset that applies the hotword rule to the PERSON_NAME infotype.
+	var ruleSet = &dlppb.InspectionRuleSet{
+		InfoTypes: []*dlppb.InfoType{
+			{Name: infoTypeName}, //"PERSON_NAME"
+		},
+		Rules: []*dlppb.InspectionRule{
+			{
+				Type: &dlppb.InspectionRule_HotwordRule{
+					HotwordRule: hotwordRule.HotwordRule,
+				},
+			},
+		},
+	}
+
+	// Construct the Inspect request to be sent by the client.
+	req := &dlppb.InspectContentRequest{
+		Parent: fmt.Sprintf("projects/%s/locations/global", projectID),
+		Item:   contentItem,
+		InspectConfig: &dlppb.InspectConfig{
+			InfoTypes: []*dlppb.InfoType{
+				{Name: infoTypeName}, //"PERSON_NAME"
+			},
+			IncludeQuote:  true,
+			MinLikelihood: dlppb.Likelihood_VERY_LIKELY,
+			RuleSet: []*dlppb.InspectionRuleSet{
+				ruleSet,
+			},
+		},
+	}
+
+	// Send the request.
+	resp, err := client.InspectContent(ctx, req)
+	if err != nil {
+		fmt.Fprintf(w, "Receive: %v", err)
```

Comment:
nit: you don't need to do both, write the error to the writer and return the error. Best to just return the error.

Reply:
okay, got your point.